librecval/extract_phrases.py

```diff
@@ -133,45 +158,291 @@ def extract_session(self, session_dir: Path):
+        if cree_phrase_tier
+        else []
+    )
+    for cree_phrase in cree_phrases:
+        s, sound_bite = extract_data(
+            eaf_file,
+            "sentence",
+            cree_phrase,
+            audio,
+            speaker,
+            session_id,
+            english_phrase_tier,
+            comment_tier,
+        )
+        count += 1
+        print(f"Processed audio segment number {count}")
+        yield s, sound_bite
+
+
+def get_word_tiers(keys):
+
+    english_word_tier = "English (word)" if "English (word)" in keys else None
+    cree_word_tier = "Cree (word)" if "Cree (word)" in keys else None
+
+    return english_word_tier, cree_word_tier
+
+
+def get_phrase_tiers(keys):
+
+    english_phrase_tier = "English (sentence)" if "English (sentence)" in keys else None
+    cree_phrase_tier = "Cree (sentence)" if "Cree (sentence)" in keys else None
+
+    return english_phrase_tier, cree_phrase_tier
+
+
+def extract_data(
+    _file, _type, snippet, audio, speaker, session_id, english_tier, comment_tier
+) -> tuple:
+    """
+    Extracts all relevant data from a .eaf file, where "relevant data" is:
+    - translation
+    - transcription
+    - start time
+    - stop time
+    - comment
+    - quality
+    Returns a Segment and a sound bite
+    """
+
+    start = snippet[0]
+    stop = snippet[1]
+    transcription = snippet[2]
+
+    translation = _file.get_annotation_data_at_time(english_tier, start + 1)
+    translation = (
+        translation[0][2] if len(translation) > 0 and len(translation[0]) == 3 else ""
+    )
+
+    comment = _file.get_annotation_data_at_time(comment_tier, start + 1) or ""
+    comment = comment[0][2] if len(comment) > 0 and len(comment[0]) == 3 else ""
+
+    quality = "unknown"
+    if "good" in comment.lower() or "best" in comment.lower():
+        quality = "good"
+    elif "bad" in comment.lower():
+        quality = "bad"
+
+    sound_bite = audio[start:stop]
+
+    # normalize
+    transcription = normalize(transcription)
+    translation = normalize(translation)
+    sound_bite = sound_bite.normalize(headroom=0.1)
+
+    s = Segment(
+        translation=translation,  # in English
+        transcription=transcription,  # in Cree
+        type=_type,
+        start=start,
+        stop=stop,
+        comment=comment,
+        speaker=speaker,
+        quality=quality,
+        session=session_id,
+        audio=sound_bite,
+    )
+
+    return s, sound_bite
+
+
+@logme.log
+def find_audio_oddities(annotation_path: Path, logger=None) -> Optional[Path]:
+    """
+    Finds the associated audio in Audacity's format.
+    """
+
+    # Get folder name without expected track name
+    i = str(annotation_path).rfind("/")
+    _path = str(annotation_path)[:i]
+
+    # the track number is between the last - and the last .
+    j = str(annotation_path).rfind("-")
+    k = str(annotation_path).rfind(".")
+    track = str(annotation_path)[j + 1 : k]
+    track_1 = track.split("_")[0]
```

Comment:
I actually don't mind using regex, but it can be harder to understand at a glance 🤔 @eddieantonio which do you prefer?

Reply:
There are also places later on where having the k index is useful:
Line 420: `track_8 = str(annotation_path)[i + 1 : k]` to get just the filename (no extension)
Line 437: `track_13 = str(annotation_path)[i + 1 : k]` again, to get just the filename (no extension)

I'm sure there's a Pathlib way of doing this that would be cleaner, so please teach me!

Reply:
Regex is fine, but I like using verbose patterns like the ones found in [`get_mic_id()`](https://github.com/UAlbertaALTLab/recording-validation-interface/blob/6d398d4ccd0c7294cc90965b1aa8277a29432702/librecval/extract_phrases.py#L508-L547), because regex is so difficult to read and maintain!

To get the filename with no extension with Path, use [`annotation_path.stem`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem)